### ISSUE TITLE
Partial incompatibility with Asterisk 11.12

### ISSUE THREAD

**Issue by snarpatroid** [FreePBX/hotelwakeup#63]
Expanding on topic at http://pbxinaflash.com/community/index.php?threads/wake-up-call-module-development.16862/

Exactly the same issue as reported here http://github.com/POSSA/Hotel-Style-Wakeup-Calls/issues/62#issuecomment-77441815

Base functionality generally works but I cannot cancel or amend a call once set from an extension, I have to log onto the interface to cancel or amend a wake-up call once set.

The latest beta sounds great but this broke my FreePBX login so I had to revert.

Good luck with a fix this is an amazing module.

**Comment by lgaetz** [comment]
Testing on a system running Asterisk 11.17.1 and Freepbx 12, cannot reproduce this issue. At this point it is reported by two users (@laptopu and @snarpatroid), if anyone has something to add, please add to the ticket

I will keep looking

**Comment by lgaetz** [comment]
By default this module logs wakeup call activity in the file:
`/var/log/asterisk/wakeup.log`
Try triggering a failure and then check the contents of this file. Here is part of the log file content on a normal working call where a wuc is set and the system reads the wuc time back to you:

```
ANSWER 
      READ=200 result=0
1=0, 2=, 3=, 4=
STREAM FILE hello "" 
      READ=200 result=0 endpos=7200
1=0, 2= endpos=7200, 3=, 4=
Checking Directory [/var/spool/asterisk/outgoing] Check=[ext.107.call]
File=..
File=.
File=wuc.1430532000.ext.107.call
wakeup found=wuc.1430532000.ext.107.call saying time 2300
Wakeup time is set to 1100
STREAM FILE rqsted-wakeup-for "" 
      READ=200 result=0 endpos=20800
1=0, 2= endpos=20800, 3=, 4=
SAY NUMBER 11 ""
      READ=200 result=0
1=0, 2=, 3=, 4=
STREAM FILE digits/oclock "" 
      READ=200 result=0 endpos=7040
```

**Comment by snarpatroid** [comment]
Here is a log, it's pretty much the same as yours.  What I have noticed though is the basics look right but there's no 'goodbye' after setting the initial call.

Then, and this bit isn't in the log, if I dial back once the wuc has been set, I'd expect to get an option to either cancel or amend the wuc.  I don't, it just says 'you have requested a w-u-c for xx:xx' then clears.  I definitely had this option with earlier versions of Asterisk (not sure of all of them but 11.2.1 and 11.3).  It was only when I moved to 11.12.0 that the problem started.

Is there some way of forcing a log to show what's happening when I call back to wuc once a call has been set?

Does this help?

<code>
---Start of log---
--- Start of Asterisk channel data ---
    agi_request: wakeupphp
    agi_channel: SIP/202-000001a9
    agi_language: en
    agi_type: SIP
    agi_uniqueid: 1430763206.522
    agi_version: 11.12.0
    agi_callerid: 202
    agi_calleridname: Extension 202
    agi_callingpres: 1
    agi_callingani2: 0
    agi_callington: 0
    agi_callingtns: 0
    agi_dnid: 868
    agi_rdnis: unknown
    agi_context: from-internal
    agi_extension: 868
    agi_priority: 4
    agi_enhanced: 0.0
    agi_accountcode: 
    agi_threadid: -1281411984
    
--- End of Asterisk channel data ---
Channel: SIP/202-000001a9
Channel SPLIT-STA: 202 - CHAN: SIP
ANSWER 
      READ=200 result=0
1=0, 2=, 3=, 4=
STREAM FILE hello "" 
      READ=200 result=-1 endpos=0
1=-1, 2= endpos=0, 3=, 4=
Checking Directory [/var/spool/asterisk/outgoing] Check=[ext.202.call]
File=wuc.1430763240.ext.202.call
File=..
File=.
wakeup found=wuc.1430763240.ext.202.call saying time 1914
Wakeup time is set to 0714
STREAM FILE rqsted-wakeup-for "" 
      READ=200 result=0 endpos=18387
1=0, 2= endpos=18387, 3=, 4=
SAY NUMBER 7 ""
      READ=200 result=0
1=0, 2=, 3=, 4=
SAY NUMBER 14 ""
      READ=200 result=0
1=0, 2=, 3=, 4=
STREAM FILE digits/p-m "" 
      READ=200 result=0 endpos=5654
1=0, 2= endpos=5654, 3=, 4=
</code>
